### ISSUE TITLE
getting loss as NAN during training for 7k crops

### ISSUE THREAD

**Issue by anidiatm41** [GitYCC/rcnn-pytorch#7]
Desktop\ptr_torch6\src>python train.py
device: cpu
epoch: 1
train_batch_loss[ 10 ]:  nan
train_batch_loss[ 20 ]:  nan
train_batch_loss[ 30 ]:  nan
train_batch_loss[ 40 ]:  nan
train_batch_loss[ 50 ]:  nan
train_batch_loss[ 60 ]:  nan
train_batch_loss[ 70 ]:  nan
Corrupted image for 19

**Comment by chunplusplus** [comment]
Similar problem here...

Evaluate: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 540/540 [09:36<00:00,  1.07s/it]
valid_evaluation: loss=1.832921710359357, acc=0.7594752661693344
train_batch_loss[ 6510 ]:  1.5743603706359863
train_batch_loss[ 6520 ]:  2.0356717109680176
train_batch_loss[ 6530 ]:  2.1139347553253174
train_batch_loss[ 6540 ]:  1.479076623916626
train_batch_loss[ 6550 ]:  1.7650470733642578
train_batch_loss[ 6560 ]:  1.500727891921997
train_batch_loss[ 6570 ]:  1.6885172128677368
train_batch_loss[ 6580 ]:  1.9712326526641846
train_batch_loss[ 6590 ]:  nan
train_batch_loss[ 6600 ]:  nan
train_batch_loss[ 6610 ]:  nan
train_batch_loss[ 6620 ]:  nan
train_batch_loss[ 6630 ]:  nan
train_batch_loss[ 6640 ]:  nan
train_batch_loss[ 6650 ]:  nan

**Comment by GitYCC** [comment]
@chunplusplus @anidiatm41 

it might be gradient explosion, try to put 
`torch.nn.utils.clip_grad_norm_(crnn.parameters(), 5)  # gradient clipping with 5` 
right before `optimizer.step()` (https://github.com/GitYCC/crnn-pytorch/blob/master/src/train.py#L30)

I expect your feedback, thank you!

**Comment by chunplusplus** [comment]
same result...

train_batch_loss[ 2510 ]:  2.182279586791992
train_batch_loss[ 2520 ]:  3.0501251220703125
train_batch_loss[ 2530 ]:  1.902832269668579
train_batch_loss[ 2540 ]:  2.546463966369629
train_batch_loss[ 2550 ]:  2.7326841354370117
train_batch_loss[ 2560 ]:  2.863603115081787
train_batch_loss[ 2570 ]:  2.9436488151550293
train_batch_loss[ 2580 ]:  nan
train_batch_loss[ 2590 ]:  nan
train_batch_loss[ 2600 ]:  nan
train_batch_loss[ 2610 ]:  nan

**Comment by GitYCC** [comment]
@chunplusplus 
try to update `criterion = CTCLoss(reduction='sum', zero_infinity=True)` (https://github.com/GitYCC/crnn-pytorch/blob/master/src/train.py#L81)

thanks

**Comment by chunplusplus** [comment]
The problem was fixed by this update, thanks!
